mozilla_aws_cli/static/index.js

```diff
@@ -68,7 +70,27 @@ const shutdown = async () => {
 };
 
 const pollState = setInterval(async () => {
+    if (! state.backendInProgress) {
+        state.backendInProgress = true;
+    } else {
+        // The previous pollState async has not yet finished, skipping this async execution
+        return false;
+    }
     const id = new URLSearchParams(window.location.search).get("state").split("-")[0];
+    if (! state.heartbeatRunning) {
+        state.heartbeatRunning = true;
+        fetch(`/api/heartbeat?id=${id}`, {
+            method: "GET"
+        }).then((response) => {
```

Comment:
If you're making calls into fetch and using `.then()`, you should use `await` instead. They are both synchronously waiting for a Promise to resolve, and `await` makes the code far easier to read.